Implicit conversion warnings in SSE intrinsics

When compiling with Zig or modern Clang compilers, warnings are generated from libde265/libde265/x86/sse-motion.cc:
```
warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1
[-Wconstant-conversion]
_mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
```

Steps to reproduce
```
> export CGO_ENABLED=1
> export CC="zig cc -target x86_64-linux-musl"
> export CXX="zig c++ -target x86_64-linux-musl"
> go test ./...
# github.com/jdeng/goheif/libde265
In file included from libde265.cc:2:
In file included from ./libde265-all.inl:47:
libde265/libde265/x86/sse-motion.cc:3530:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3530 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:3530:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3530 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:3530:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3530 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:3530:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3530 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:3532:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3532 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:3532:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3532 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:3532:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3532 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:3532:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3532 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:3713:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3713 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:3713:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3713 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:3713:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3713 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:3713:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3713 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:3715:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3715 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:3715:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3715 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:3715:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3715 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:3715:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3715 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:3890:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3890 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:3890:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3890 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:3890:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3890 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:3890:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3890 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:3892:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3892 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:3892:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3892 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:3892:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3892 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:3892:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 3892 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4061:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4061 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4061:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4061 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4061:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4061 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4061:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4061 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4063:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4063 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4063:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4063 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4063:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4063 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4063:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4063 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4244:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4244 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4244:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4244 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4244:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4244 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4244:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4244 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4246:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4246 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4246:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4246 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4246:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4246 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4246:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4246 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4422:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4422 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4422:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4422 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4422:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4422 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4422:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4422 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4424:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4424 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4424:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4424 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4424:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4424 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4424:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4424 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4595:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4595 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4595:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4595 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4595:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4595 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4595:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4595 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4597:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4597 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4597:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4597 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4597:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4597 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4597:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4597 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4781:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4781 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4781:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4781 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4781:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4781 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4781:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4781 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4783:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4783 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4783:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4783 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4783:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4783 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4783:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4783 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4961:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4961 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4961:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4961 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4961:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4961 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4961:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4961 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
libde265/libde265/x86/sse-motion.cc:4963:68: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4963 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                                  ^~~~~
libde265/libde265/x86/sse-motion.cc:4963:58: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4963 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~                        ^~~~~
libde265/libde265/x86/sse-motion.cc:4963:48: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4963 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~              ^~~~~
libde265/libde265/x86/sse-motion.cc:4963:38: warning: implicit conversion from 'int' to 'short' changes value from 65535 to -1 [-Wconstant-conversion]
 4963 |                     _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
      |                     ~~~~~~~~~~~~~    ^~~~~
72 warnings generated.
ok  	github.com/jdeng/goheif	0.129s
```

--- a/libde265/libde265/x86/sse-motion.cc
+++ b/libde265/libde265/x86/sse-motion.cc
@@ -3527,9 +3527,9 @@ void ff_hevc_put_hevc_qpel_h_1_v_1_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -3710,9 +3710,9 @@ void ff_hevc_put_hevc_qpel_h_1_v_2_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -3887,9 +3887,9 @@ void ff_hevc_put_hevc_qpel_h_1_v_3_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -4058,9 +4058,9 @@ void ff_hevc_put_hevc_qpel_h_2_v_1_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -4241,9 +4241,9 @@ void ff_hevc_put_hevc_qpel_h_2_v_2_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -4419,9 +4419,9 @@ void ff_hevc_put_hevc_qpel_h_2_v_3_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -4592,9 +4592,9 @@ void ff_hevc_put_hevc_qpel_h_3_v_1_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -4778,9 +4778,9 @@ void ff_hevc_put_hevc_qpel_h_3_v_2_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 
@@ -4958,9 +4958,9 @@ void ff_hevc_put_hevc_qpel_h_3_v_3_sse(int16_t *dst, ptrdiff_t dststride,
             r0 = _mm_srli_epi32(r0, 6);
 
             r1 = _mm_and_si128(r1,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_and_si128(r0,
-                    _mm_set_epi16(0, 65535, 0, 65535, 0, 65535, 0, 65535));
+                    _mm_set_epi16(0, -1, 0, -1, 0, -1, 0, -1));
             r0 = _mm_hadd_epi16(r0, r1);
             _mm_store_si128((__m128i *) &dst[x], r0);
 

